staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go

```diff
@@ -630,6 +630,90 @@ func TestRunExposeService(t *testing.T) {
 			expected: "namespace: testns",
 			status:   200,
 		},
+		{
+			name: "multiple-nodeports",
+			args: []string{"service", "baz"},
+			ns:   "test",
+			calls: map[string]string{
+				"GET":  "/namespaces/test/services/baz",
+				"POST": "/namespaces/test/services",
+			},
+			input: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "baz", Namespace: "test", ResourceVersion: "12"},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{"app": "go"},
+				},
+			},
+			flags: map[string]string{"selector": "func=stream", "protocol": "UDP", "port": "80,81", "name": "foo", "labels": "svc=test", "type": "NodePort", "node-port": "30080,30081", "dry-run": "client", "output": "yaml"},
+			output: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "", Labels: map[string]string{"svc": "test"}},
+				Spec: corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       80,
+							TargetPort: intstr.FromInt(80),
+							NodePort:   30080,
+						},
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       81,
+							TargetPort: intstr.FromInt(81),
+							NodePort:   30081,
+						},
+					},
+					Selector: map[string]string{"func": "stream"},
+					Type:     corev1.ServiceTypeNodePort,
+				},
+			},
+			expected: "  ports:\n  - name: port-1\n    nodePort: 30080\n    port: 80\n    protocol: UDP\n    targetPort: 80\n  - name: port-2\n    nodePort: 30081\n    port: 81\n    protocol: UDP\n    targetPort: 81",
+			status:   200,
+		},
+		{
+			name: "multiple-nodeports-less-than-ports",
+			args: []string{"service", "baz"},
+			ns:   "test",
+			calls: map[string]string{
+				"GET":  "/namespaces/test/services/baz",
+				"POST": "/namespaces/test/services",
+			},
+			input: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "baz", Namespace: "test", ResourceVersion: "12"},
+				Spec: corev1.ServiceSpec{
+					Selector: map[string]string{"app": "go"},
+				},
+			},
+			flags: map[string]string{"selector": "func=stream", "protocol": "UDP", "port": "80,81,82", "name": "foo", "labels": "svc=test", "type": "NodePort", "node-port": "30080,30081"},
+			output: &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "foo", Namespace: "", Labels: map[string]string{"svc": "test"}},
+				Spec: corev1.ServiceSpec{
+					Ports: []corev1.ServicePort{
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       80,
+							TargetPort: intstr.FromInt(80),
+							NodePort:   30080,
+						},
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       81,
+							TargetPort: intstr.FromInt(81),
+							NodePort:   30081,
+						},
+						{
+							Protocol:   corev1.ProtocolUDP,
+							Port:       82,
+							TargetPort: intstr.FromInt(82),
+							NodePort:   31082,
```

Comment:
the port will be randomly allocated by the apiserver if empty

Reply:
in this case I don't think that the cli should be smart and assume the user wants the net, it may happen the nodeports are not consecutively.
I think that it should be left empty or just fail, maybe fail since the user is specifying node-ports explicitly, so we can assume he forgot to set it, otherwise is not normal to set only ones and left others

Reply:
On ```kubectl create service```, if target port is absent it is assumed to be default (same as port).
```
$ kubectl create service nodeport test-nodeport --tcp=80:8080 --tcp=81 --dry-run=client -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    app: test-nodeport
  name: test-nodeport
spec:
  ports:
  - name: 80-8080
    port: 80
    protocol: TCP
    targetPort: 8080
  - name: "81"
    port: 81
    protocol: TCP
    targetPort: 81
  selector:
    app: test-nodeport
  type: NodePort
status:
  loadBalancer: {}
```
So I think it is natural that node port assume default when it is absent.
How do you think ?

Reply:
yeah, but I refer to this case
```
, "port": "80,81,82",  "node-port": "30080,30081"
```
I don't see that is correct default to 30082, what happens if node-port is: "30080,31223", do you use 31224?
Also, nodeports can be set by the user or otherwise they are randomly allocated by the apiserver, so the default should be to left it empty.
But my point is that in case that the user specify multi-ports, we should validate that the number of ports and node-port match, because the user is explicitly saying he wants "those" multi-ports, so it feels weird to say I want this specific nodeports but just giveme some other random :)

Reply:
> I don't see that is correct default to 30082, what happens if node-port is: "30080,31223", do you use 31224?

Do not worry about it, I understand the third node-port value will be assinged randomly.
I just use 31082 because I need some number.
I will remove this case, but I want to know how to write test case for such a random value.

> But my point is that in case that the user specify multi-ports, we should
> validate that the number of ports and node-port match, because the user is
> explicitly saying he wants "those" multi-ports, so it feels weird to say I
> want this specific nodeports but just giveme some other random :)

Ok, I will fix this.